pkg/dns/azure/client/client.go

```diff
@@ -141,45 +144,49 @@ func (c *recordSetClient) Delete(ctx context.Context, zone Zone, arec ARecord) e
 }
 
 type privateRecordSetClient struct {
-	client privatedns.RecordSetsClient
+	client *armprivatedns.RecordSetsClient
 }
 
-func newPrivateRecordSetClient(config Config, userAgentExtension string) (*privateRecordSetClient, error) {
-	authorizer, err := getAuthorizerForResource(config)
+func newPrivateRecordSetClient(config Config) (*privateRecordSetClient, error) {
+	_, cred, err := getAuthorizerForResource(config)
 	if err != nil {
 		return nil, err
 	}
 
-	prc := privatedns.NewRecordSetsClientWithBaseURI(config.Environment.ResourceManagerEndpoint, config.SubscriptionID)
-	prc.AddToUserAgent(userAgentExtension)
-	prc.Authorizer = authorizer
+	cloudConfig := ParseCloudEnvironment(config)
+	options := &arm.ClientOptions{
+		ClientOptions: policy.ClientOptions{
+			Cloud: cloudConfig,
+		},
+	}
+
+	prc, err := armprivatedns.NewRecordSetsClient(config.SubscriptionID, cred, options)
+	if err != nil {
+		return nil, err
+	}
 	return &privateRecordSetClient{client: prc}, nil
 }
 
 func (c *privateRecordSetClient) Put(ctx context.Context, zone Zone, arec ARecord, metadata map[string]*string) error {
-	rs := privatedns.RecordSet{
-		RecordSetProperties: &privatedns.RecordSetProperties{
+	rs := armprivatedns.RecordSet{
+		Properties: &armprivatedns.RecordSetProperties{
 			TTL: &arec.TTL,
-			ARecords: &[]privatedns.ARecord{
-				{Ipv4Address: &arec.Address},
+			ARecords: []*armprivatedns.ARecord{
+				{IPv4Address: &arec.Address},
 			},
 			Metadata: metadata,
 		},
 	}
-	_, err := c.client.CreateOrUpdate(ctx, zone.ResourceGroup, zone.Name, privatedns.A, arec.Name, rs, "", "")
+
+	_, err := c.client.CreateOrUpdate(ctx, zone.ResourceGroup, zone.Name, armprivatedns.RecordTypeA, arec.Name, rs, nil)
 	if err != nil {
 		return errors.Wrapf(err, "failed to update dns a record: %s.%s", arec.Name, zone.Name)
 	}
 	return nil
 }
 
 func (c *privateRecordSetClient) Delete(ctx context.Context, zone Zone, arec ARecord) error {
-	_, err := c.client.Get(ctx, zone.ResourceGroup, zone.Name, privatedns.A, arec.Name)
-	if err != nil {
-		// TODO: How do we interpret this as a notfound error?
-		return nil
-	}
-	_, err = c.client.Delete(ctx, zone.ResourceGroup, zone.Name, privatedns.A, arec.Name, "")
+	_, err := c.client.Delete(ctx, zone.ResourceGroup, zone.Name, armprivatedns.RecordTypeA, arec.Name, nil)
```

Comment:
This might change the behavior in some failure cases (in particular if the operator failed to publish the record initially and is now trying to delete it).  We'll need to watch for that.